### PR TITLE
Callbacks vnext

### PR DESCRIPTION
**Pull request by danielmarbach**
Relates to https://github.com/Particular/NServiceBus.Callbacks/pull/62

### CODE REVIEW

**Comment by danielmarbach** [comment]
@Particular/docs-maintainers where is the best place to add the upgrade guide (menu and folder)?

**Comment by SzymonPobiega** [comment]
@danielmarbach how about `https://docs.particular.net/nservicebus/upgrades/callbacks-1to2` (similar to `https://docs.particular.net/nservicebus/upgrades/sqlpersistence-1to2`)?

**Comment by danielmarbach** [comment]
I think this is ready for an initial review @Particular/nservicebus-maintainers

File: nservicebus/messaging/callbacks_enum-reply_callbacks_[2,).partial.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+NOTE: When replying with an enum value, the replying endpoint also needs to reference the `NServiceBus.Callbacks` package.
+
+If the endpoint only replies to callbacks use the reply-only mode as shown below:
```
Comment (timbussmann) [inline nservicebus/messaging/callbacks_enum-reply_callbacks_[2,).partial.md]: you call it 'reply-only' mode, but you named the config option `makesRequests` which doesn't align very well imho (I prefer 'reply-only')

File: nservicebus/upgrades/callbacks-1to2.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+reviewed: 2017-05-05
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 6
+---
+
+
+## Enable callbacks
+
+Callbacks have been made opt-in. In order to make callbacks work they have to be enabled.
```
Comment (timbussmann) [inline nservicebus/upgrades/callbacks-1to2.md]: they have to be enabled explicitly

**Pull request merged**